core/src/main/java/io/grpc/NameResolver.java

```diff
@@ -350,4 +431,143 @@ public String toString() {
       }
     }
   }
+
+  /**
+   * Represents the results from a Name Resolver.
+   *
+   * @since 1.20.0
+   */
+  @ExperimentalApi("FIXME")
+  public static final class Result {
+    private final List<EquivalentAddressGroup> servers;
+    @ResolutionResultAttr
+    private final Attributes attributes;
+    @Nullable
+    private final Object serviceConfig;
+
+    Result(
+        List<EquivalentAddressGroup> servers,
+        @ResolutionResultAttr Attributes attributes,
+        Object serviceConfig) {
+      this.servers = Collections.unmodifiableList(new ArrayList<>(servers));
+      this.attributes = checkNotNull(attributes, "attributes");
+      this.serviceConfig = serviceConfig;
+    }
+
+    /**
+     * Constructs a new builder of a name resolution result.
+     *
+     * @since 1.20.0
+     */
+    public static Builder newBuilder() {
+      return new Builder();
+    }
+
+    /**
+     * Gets the servers resolved by name resolution.
+     *
+     * @since 1.20.0
+     */
+    public List<EquivalentAddressGroup> getServers() {
+      return servers;
+    }
+
+    /**
+     * Gets the attributes associated with the servers resolved by name resolution.
+     *
+     * @since 1.20.0
+     */
+    @ResolutionResultAttr
```

Comment:
Maybe rename `Result` to `ResolutionResult` to match the annotation name.

Reply:
Done.